src/app/zap-templates/zcl/data-model/chip/tls-client-management-cluster.xml

```diff
@@ -0,0 +1,96 @@
+  <domain name="General"/>
+  <enum name="TLSEndpointStatusEnum" type="enum8">
+    <cluster code="0x0802"/>
+    <item name="Provisioned" value="0x00"/>
+    <item name="InUse" value="0x01"/>
+  </enum>
+
+  <struct name="TLSEndpointStruct" apiMaturity="provisional">
+    <cluster code="0x0802"/>
+    <item fieldId="0" name="EndpointID" type="int16u"/>
+    <item fieldId="1" name="Hostname" type="octet_string"/>
+    <item fieldId="2" name="Port" type="int16u"/>
+    <item fieldId="3" name="CAID" type="int16u"/>
+    <item fieldId="4" name="CCDID" type="int16u" isNullable="true" optional="true"/>
+    <item fieldId="5" name="Status" type="TLSEndpointStatusEnum" min="0x00" max="0x01"/>
+  </struct>
+
+  <cluster apiMaturity="provisional">
+    <domain name="General"/>
+    <name>TLS Client Management</name>
+    <code>0x0802</code>
+    <define>TLS_CLIENT_MANAGEMENT_CLUSTER</define>
+    <description>This Cluster is used to provision TLS Endpoints with enough information to facilitate subsequent connection.</description>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <globalAttribute code="0xFFFD" side="either" value="1"/>
+    <attribute code="0x0000" side="server" define="MAX_PROVISIONED" type="int8u">MaxProvisioned</attribute>
+    <attribute code="0x0001" side="server" define="CURRENT_PROVISIONED" type="int8u">CurrentProvisioned</attribute>
+    <attribute code="0x0002" side="server" define="MAX_IN_USE" type="int8u">MaxInUse</attribute>
+    <attribute code="0x0003" side="server" define="CURRENT_IN_USE" type="int8u">CurrentInUse</attribute>
+    <command code="0x00" source="client" name="ProvisionEndpoint" optional="false" response="ProvisionEndpointResponse">
+      <description>This command SHALL provision a TLS Endpoint for the provided HostName / Port combination.</description>
+      <access op="invoke" privilege="administer"/>
+      <arg id="0" name="Hostname" type="octet_string"/>
+      <arg id="1" name="Port" type="int16u"/>
+      <arg id="2" name="CAID" type="int16u"/>
+      <arg id="3" name="CCDID" type="int16u" optional="true" isNullable="true"/>
+      <arg id="4" name="EndpointID" type="int16u" optional="true" isNullable="true"/>
+    </command>
+
+    <command code="0x01" source="server" name="ProvisionEndpointResponse" optional="false" disableDefaultResponse="true">
+      <description>This command SHALL be generated in response to a ProvisionEndpointRequest command.</description>
+      <arg id="0" name="EndpointID" type="int16u"/>
+    </command>
+
+    <command code="0x02" source="client" name="FindEndpoint" optional="false" response="FindEndpointResponse">
+      <description>This command SHALL return the TLS Endpoint details for the passed in EndpointID.</description>
+      <arg id="0" name="EndpointID" type="int16u" optional="true" isNullable="true"/>
```

Comment:
This is not optional in the spec.

Reply:
Updating PR, Aaron made some changes a few days ago @ https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10794